docs/basic/divide-and-conquer.md

```diff
@@ -1,88 +1,163 @@
 author: fudonglai, AngelKitty, labuladong
 
-首先简单阐述一下递归，分治算法，动态规划，贪心算法这几个东西的区别和联系，心里有个印象就好。
+Last translated with upstream [08eb0c5](https://github.com/OI-wiki/OI-wiki/commit/08eb0c53b1ad55b518b0f5f7a569c25b8c1215b2#diff-f6f59ea091666854504b33243563a98ec55edf8f4c6a26dec1b2726479de41b4)
 
-递归是一种编程技巧，一种解决问题的思维方式；分治算法和动态规划很大程度上是递归思想基础上的（虽然实现动态规划大都不是递归了，但是我们要注重过程和思想），解决更具体问题的两类算法思想；贪心算法是动态规划算法的一个子集，可以更高效解决一部分更特殊的问题。
+The article will introduce difference and application of recursive algorithm and divide-and-conquer algorithm
 
-分治算法将在这节讲解，以最经典的归并排序为例，它把待排序数组不断二分为规模更小的子问题处理，这就是“分而治之”这个词的由来。显然，排序问题分解出的子问题是不重复的，如果有的问题分解后的子问题有重复的（重叠子问题性质），那么这就交给动态规划算法去解决！
+## Introduction
 
-## 递归详解
+In computer science and math, recursion is method by using functions that call themselves from within their own definition. Additionally in computer science it also refers to a method of solving problem by repeatedly decomposing problem into sub-problems of the same type as the original.
 
-介绍分治之前，首先要弄清楚递归这个概念。
+From [Wikipedia](https://en.wikipedia.org/wiki/Divide-and-conquer_algorithm):
+> In computer science, divide and conquer is an algorithm design paradigm. A divide-and-conquer algorithm recursively breaks down a problem into two or more sub-problems of the same or related type, until these become simple enough to be solved directly. The solutions to the sub-problems are then combined to give a solution to the original problem. <!---中文显然就是抄英文wiki的-->
 
-递归的基本思想是某个函数直接或者间接地调用自身，这样就把原问题的求解转换为许多性质相同但是规模更小的子问题。我们只需要关注如何把原问题划分成符合条件的子问题，而不需要去研究这个子问题是如何被解决的。递归和枚举的区别在于：枚举是横向地把问题划分，然后依次求解子问题，而递归是把问题逐级分解，是纵向的拆分。
+## Detailed Introduction
 
-以下会举例说明我对递归的一点理解， **如果你不想看下去了，请记住这几个问题怎么回答：** 
+### Recursion
 
-1. 如何给一堆数字排序？答：分成两半，先排左半边再排右半边，最后合并就行了，至于怎么排左边和右边，请重新阅读这句话。
-2. 孙悟空身上有多少根毛？答：一根毛加剩下的毛。
-3. 你今年几岁？答：去年的岁数加一岁，1999 年我出生。
+> To understand recursion, you must first understand recursion.
 
-递归代码最重要的两个特征：结束条件和自我调用。自我调用是在解决子问题，而结束条件定义了最简子问题的答案。
+The fundamental idea of recursion is a function call itself directly or indirectly, so that the original problem transforms into many smaller size sub-problems with similar properties. While solving problems it only needs to focus on how to decompose original problems to eligible sub-problems without too much attention about how it is solved.
```

Comment:
```suggestion
The fundamental idea of recursion is a function calling itself directly or indirectly, so that the original problem transforms into many sub-problems with similar properties but smaller size. When solving problems, we only need to focus on how to decompose original problems to proper sub-problems, but not too much attention on how it is solved.
```